dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/snowflake/SnowflakeMetadataConnector.java

```diff
@@ -148,16 +155,30 @@ protected void addSqlTasksWithInfoSchemaFallback(
     }
   }
 
-  @ForOverride
-  protected void addSingleSqlTask(
+  private void addSingleSqlTask(
       @Nonnull List<? super Task<?>> out,
-      @Nonnull Class<? extends Enum<?>> header,
       @Nonnull String format,
-      @Nonnull TaskVariant task) {
+      @Nonnull TaskVariant task,
+      @Nonnull Function<ResultSet, String[]> transformer) {
     out.add(
         new JdbcSelectTask(
                 task.zipEntryName, String.format(format, task.schemaName, task.whereClause))
-            .withHeaderClass(header));
+            .withHeaderTransformer(transformer));
+  }
+
+  @CheckReturnValue
+  private String[] transformHeaderToCamelCase(ResultSet rs, CaseFormat baseFormat) {
+    try {
+      ResultSetMetaData metaData = rs.getMetaData();
+      int columnCount = metaData.getColumnCount();
+      String[] columns = new String[columnCount];
+      for (int i = 0; i < columnCount; i++) {
+        columns[i] = baseFormat.to(CaseFormat.UPPER_CAMEL, metaData.getColumnLabel(i + 1));
+      }
+      return columns;
+    } catch (SQLException e) {
```

Comment:
I don't particularly like catching and discarding exceptions, so I went on an adventure:

HeaderTransformer is called in a place where SQLException may be thrown.
So what if we define a custom functional interface with `String[] apply(ResultSet) throws SQLException` and use that instead of `Function<ResultSet, String[]>`? Then this try/catch can be removed entirely, and this method can just throw.

Reply:
This feels much cleaner. Thanks!